lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc

```diff
@@ -382,87 +382,132 @@ function paraneue_dosomething_preprocess_node_campaign(&$vars) {
     drupal_add_js($js, 'inline');
   }
 
+  // Setup the reportback showcase.
+  paraneue_dosomething_preprocess_reportback_showcase($vars);
+
   // If this is not the pitch page:
   if (!isset($vars['is_pitch_page'])) {
     // Set the number of files to display in initial gallery.
-    $vars['reportbacks']['initial_count'] = 6;
+    $vars['reportbacks_gallery']['desired_count'] = 6;
     // Adds themed Reportback Files.
-    paraneue_dosomething_preprocess_reportback_files($vars);
+    paraneue_dosomething_preprocess_reportback_gallery($vars);
   }
 
 }
 
+
 /**
- * Adds $gallery_items array of themed Reportback Files for given node preprocess $vars.
+ * Get total number of Reportbacks by status (promoted, approved, etc).
+ *
+ * @param $nid  Campaign node id.
+ * @param $status Reportback status (promoted, approved, etc).
+ * @return int
  */
-function paraneue_dosomething_preprocess_reportback_files(&$vars) {
-  $reportbacks_gallery = &$vars['reportbacks_gallery'];
+function paraneue_dosomething_preprocess_reportbacks_total_by_status($nid, $status) {
```

Comment:
This is minor, but the function name throws me off because i'm expecting this to preprocess an input array of variables.

I can see us using this helper function all over the place, not just in the theme. What do you think about renaming moving it into the `dosomething_reportback.module` file?